src/Adapters/DatabaseApproverRepository.php

```diff
@@ -75,11 +76,24 @@ public function setApproverCategories( int $userId, array $categoryNames ): void
 	}
 
 	private function serializeCategories( array $categories ): string {
-		return implode( '|', $categories );
+		return implode( '|', array_map(
+			fn ( string $category ) => $this->normalizeCategoryTitle( $category ),
+			$categories
+		) );
+	}
+
+	private function normalizeCategoryTitle( string $title ): string {
+		// TODO: TitleValue guaranteed no DB access:
+		// return TitleValue::tryNew( NS_CATEGORY, trim( $title ) )?->getText() ?? '';
+		// TODO: Title can possibly access DB, but unlikely here:
+		return Title::newFromText( $title )?->getText() ?? '';
 	}
 
 	private function deserializeCategories( string $serializedCategories ): array {
-		return $serializedCategories === '' ? [] : explode( '|', $serializedCategories );
+		return $serializedCategories === '' ? [] : array_map(
+			fn ( string $category ) => $this->normalizeCategoryTitle( $category ),
```

Comment:
Why do we need to normalize if we are deserializing something that is already normalized?

Reply:
I just added both as examples. Normalization on save should be enough. Unless there's a reason to normalize on retrieval instead.

Reply:
I do not see a reason to do it on read, and I'm scared about performance implications.